### ISSUE TITLE
Bugs: multiple Inputs with clearable 

### ISSUE THREAD

**Issue by ywu151** [uber-web/baseui#1643]
## Current Behavior
If there are multiple inputs with clearable, only the last input will be cleared no mater which clear icon we click

Follow the code here.
```
import React, {useState} from 'react';

import {Block} from 'baseui/block';
import {Input, SIZE as INPUT_ZISE} from 'baseui/input';

export default () => {
  const [name, setName] = useState(undefined);
  const [submitter, setSubmitter] = useState(undefined);
  return (
   <div>
     <Input
          onChange={e => setName(e.target.value)}
          placeholder="Simulation Name"
          size={INPUT_ZISE.compact}
          value={name}
          clearable
        />
        <Input
          placeholder="Submitter"
          onChange={e => setSubmitter(e.target.value)}
          size={INPUT_ZISE.compact}
          value={submitter}
          clearable
        />
     </div>
  );
};
```

## Your Environment

| Tech    | Version |
| ------- | ------- |
| Base UI | v8.11.0  |
| React   |         |
| browser |         |

- [ ] I have searched the [issues](https://github.com/uber-web/baseui/issues) of this repository and believe that this is not a duplicate.

**Comment by sandgraham** [comment]
@ywu151 Still looking at exactly why this doesn't function as expected but a workaround is to pass a ref to each `Input` via the `inputRef` prop. There is something wrong with the default `ref` used but I need to test a couple of things before releasing a patch.

Here is a sandbox showing how to get around it for now with `inputRef`: https://codesandbox.io/s/clearable-input-with-explicit-inputref-e5toh

**Comment by sandgraham** [comment]
@ywu151 This should be fixed in `v8.12.0` if you can bump your project.

**Comment by ywu151** [comment]
Great!
